src/components/Missions.js

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Table, Container, Badge, Button,
+} from 'react-bootstrap';
+import {
+  fetchMissions,
+  toggleMission,
+} from '../redux/missions/missionActionCreator';
+import { missions } from '../redux/missions/mission';
+
+const Missions = () => {
+  const dispatch = useDispatch();
+  const allMissions = useSelector(missions);
+
+  useEffect(() => {
+    dispatch(fetchMissions);
+  }, [fetchMissions]);
+
+  const joinOrLeaveMission = (e) => {
+    dispatch(toggleMission({ mission_id: e.target.id }));
+  };
+
+  const missionComponents = allMissions.map((mission) => (
+    <tr key={mission.mission_id}>
+      <td>
+        <b>{mission.mission_name}</b>
+      </td>
+      <td>{mission.description}</td>
+      <td className="px-4 align-middle">
+        {mission.reserved ? (
+          <Badge className="bg-info">Active Member</Badge>
+        ) : (
+          <Badge className="bg-secondary">NOT A MEMBER</Badge>
+        )}
+      </td>
+      <td className="px-4 align-middle">
+        {mission.reserved ? (
+          <Button
+            variant="outline-danger"
+            id={mission.mission_id}
+            onClick={joinOrLeaveMission}
+          >
```

Comment:
- [ ] Rather than using the event object to target the id of the element, you may use the `missionId` directly. To pass it to your `onClick` handler you have to use a callback function  like this `onClick={() => joinOrLeaveMission(mission_id)}`